src/main/java/com/github/manolo8/darkbot/core/manager/PetManager.java

```diff
@@ -383,7 +385,10 @@ private void updateNpcLocatorList(long gearsSprite) {
         if (locatorNpcList.getSize() < oldSize && validUntil > System.currentTimeMillis()) return;
 
         validUntil = System.currentTimeMillis() + 100;
+        int oldState = locatorList.hashCode();
         locatorNpcList.sync(locatorList, Gear::new, null);
+        if (oldState != locatorList.hashCode())
```

Comment:
I've implemented the `class Updatable.Reporting` which can be used in a new `syncAndReport` method, that returns true if anything has changed, and false otherwise.

If you rebase this PR you should be able to use that instead:

```suggestion
        if (locatorNpcList.syncAndReport(locatorList, Gear::new))
```
(You will need to update Gear to support the new interface, but it shouldn't be a big deal, see https://github.com/darkbot-reloaded/DarkBot/blob/master/src/main/java/com/github/manolo8/darkbot/core/entities/Zone.java#L41 for an example)